src/dual_context.jl

```diff
@@ -41,8 +41,10 @@ end
 ChainRulesCore.mul_zero(::Zero, p::Partials) = zero(p)
 ChainRulesCore.mul_zero(p::Partials, ::Zero) = zero(p)
 
-
 @inline function overdub(ctx::TaggedCtx{T}, f, args...) where {T}
+    if length(args) > 4
```

Comment:
Comment for why this?

Reply:
just speeds things up, we don't have `frule`s longer than that.

Reply:
Add that comment to the source code